File: 455-assign-cookies/assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(s);
        Arrays.sort(g);
        int i = g.length -1;
        int j = s.length -1;
        int content = 0;
        while(i>=0 && j>=0){
            if(s[j]<g[i]){
                i--;
            }
            else{
                content += 1;
                j--;
                i--;
            }
        }
        return content;
    }
}

File: 455-assign-cookies/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        i = len(s)-1
        j = len(g) - 1
        content = 0
        while i>=0 and j>=0:
            if s[i]<g[j]:
                j -= 1
            elif s[i]>=g[j]:
                i -= 1
                j -= 1
                content += 1
        return content